lib/common.js

```diff
@@ -38,6 +38,27 @@ common.safeRequire = (moduleName) => {
   }
 };
 
+// Mixin source methods to target without overriding existing methods
+// for ES6 classes.
+//
+common.mixin = (target, source) => {
+  Object.getOwnPropertyNames(source.prototype).forEach((property) => {
+    if (!target.prototype[property]) {
+      target.prototype[property] = source.prototype[property];
+    }
+  });
+};
+
+// If last element of the array args is a function then
+// pops the array and returns that function else return null
+//
+common.extractCallback = (args) => {
+  if (args.length > 0 && typeof args[args.length - 1] === 'function') {
+    return args.pop();
```

Comment:
This will modify arrays that a user passes to the functions of jstp's public API, right?